src/render/SDL_render.c

```diff
@@ -5346,3 +5355,102 @@ bool SDL_GetRenderVSync(SDL_Renderer *renderer, int *vsync)
     }
     return true;
 }
+
+static bool DrawDebugCharacter(SDL_Renderer *renderer, float x, float y, Uint32 c, SDL_ScaleMode scaleMode)
+{
+    const Uint32 charWidth = SDL_DEBUG_TEXT_FONT_CHARACTER_SIZE;
+    const Uint32 charHeight = SDL_DEBUG_TEXT_FONT_CHARACTER_SIZE;
+    bool result;
+    Uint32 ix, iy;
+    const unsigned char *charpos;
+    Uint32 *curpos;
+    Uint8 *linepos;
+    Uint32 pitch;
+    SDL_Surface *character;
+    Uint32 ci;
+    Uint8 r, g, b, a;
+
+    // Character index in cache
+    ci = c;
+    if (ci >= SDL_DEBUG_FONT_NUM_GLYPHS) {
+        ci = (SDL_DEBUG_FONT_NUM_GLYPHS - 1);
+    }
+
+    if ((ci <= 32) || ((ci >= 127) && (ci <= 160))) {
+        return true;  // these are just completely blank chars, don't bother doing anything.
+    }
+
+    // Allocate a new cache for this renderer if needed
+    if (!renderer->debug_char_texture_cache) {
+        renderer->debug_char_texture_cache = (SDL_Texture **) SDL_calloc(SDL_DEBUG_FONT_NUM_GLYPHS, sizeof (SDL_Texture *));
+        if (!renderer->debug_char_texture_cache) {
+            return false;
+        }
+    }
+
+    // !!! FIXME: maybe just make single font atlas texture instead of a glyph cache.
```

Comment:
Yeah, I don't think the texture would be very big, we might as well do this.

Reply:
Doing this in a single row of glyphs makes the texture 2570x10 pixels, which will upset some embedded devices and low-end phones. We can split it into multiple rows, 14 glyphs per row, for a 160x170 texture, but that's a pain in the butt for debug stuff that doesn't cost anything until used.

Let's leave this as a FIXME for now and let some other poor soul tackle that later.  :)

Reply:
Ah, who am I kidding, I'm totally doing this, lol.

Reply:
lol, I knew it :D

Reply:
This is done in f032edf2f79dd28ede56759959b0dd2d573c9958. It was worth it!